### PR TITLE
Extend daemon api

### PR DESCRIPTION
**Pull request by BaderBC**

### CODE REVIEW

**Review by sewe2000 — APPROVED** [review]

**Review by maxidragon — COMMENTED** [review]
You added some API methods, can you add documentation to them?

File: pkg/api/daemon.go [hunk 1]
```diff
@@ -281,3 +282,56 @@ func GetDaemon(daemonName string) (Daemon, error) {
...
 	}
 }
+
+func errorIfNotSystemd() error {
+	initSystem, err := GetInitSystem()
+	if err != nil {
+		return err
+	}
+
+	if initSystem == SYSTEMD {
```
Comment (cooligus) [inline pkg/api/daemon.go:292]: I don't think it's good design, it's better to remove artix test

File: pkg/daemon_tracker.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	foundConflict, conflictMsg := daemonTracker.FindConflicts()
+	if foundConflict {
+		return errors.New(conflictMsg)
+	}
+
+	return runSystemdCommand("disable", daemonName)
+}
+
+// FindConflicts returns a boolean indicating if there are any conflicts and a string with more details
+func (daemonTracker *DaemonTracker) FindConflicts() (bool, string) {
```
Comment (cooligus) [inline pkg/daemon_tracker.go:81]:
```suggestion
func (daemonTracker *DaemonTracker) FindConflicts() (bool, error) {
```

File: pkg/daemon_tracker.go
Comment (cooligus) [inline pkg/daemon_tracker.go:1]: I feel this could be optimized as you create 5 functions that do literally the same

**Review by cooligus — CHANGES_REQUESTED** [review]

**Comment by cooligus** [comment]
Could you also create tests?

**Review by cooligus — APPROVED** [review]

**Pull request merged**